packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+---
+name: Spec
+menu: Table
+route: /table-spec
+---
+
+# Technical Design - Espressive Table
+
+# Current implementations
+
+As of `goldengirls` release, we use tables in the following applications:
+
+Barista:
+
+- Admin
+  - Service Department
+  - Service Teams
+  - Barista FAQs
+  - Users
+  - Locations
+
+Caffeine
+
+- Configuration
+- Entities
+- Localization
+- Training
+- Variables
+
+## Features in tables
+
+The features in those tables vary from one implementation to another, but the general idea is to present the data in a tabular way, ussually also providing the means for data persistance. The aggregated features list is the following:
+
+**Item creation**:
+Some tables allow the user to create a new entry in the table. Adding an empty row.
+
+**Inline edition**:
+Some tables allow the user to edit the data, replacing the display representation with some JSX that allows the mutation of the data.
+
+**Item deletion**:
+Some tables allow the user to delete record via a button displayed at the row level.
+
+**Custom actions**:
+In most implementations, tables add a trailing column to each row. This row can display `action buttons`, which can be mapped to corresponding actions over the entire row. i.e. edit, delete, cancel, etc.
+
+## Data representation / mutation
+
+The way each cell represents the data for either display or mutation purposes not only depends on the data type but also on business logic, in most cases, values are easily coerced to string for display purposes and mutations are easily achieved via a controlled input component. The rest of scenarios require a more complex implementation, e.g. the `employee` field in the `Users` table from `Barista Admin` which under the carpet is a single `eid`, but the representation of that cell is an avatar with the user name it points to, although it could be any other representation, specially when editing the value.
+
+# Proposal
+
+Much thought has been put to the proposal described in this section, our goal is to create a table that can support all of our current use cases while at the same time being abstract enough to support future ones. We are not inventing the wheel here, there are great open source libraries like `react-table`. The problem with those libraries is that they are bloated with things we don't need either because we already abstracted such functionality out to other components or we sinply don't need it.
```

Comment:
```suggestion
Much thought has been put to the proposal described in this section, our goal is to create a table that can support all of our current use cases while at the same time being abstract enough to support future ones. We are not inventing the wheel here, there are great open source libraries like `react-table`. The problem with those libraries is that they are bloated with things we don't need either because we already abstracted such functionality out to other components or we simply don't need it.
```

Reply:
I forgot that Github has this feature to insert a suggested change. WAY easier for some of these changes to just make the edit and suggest it that way. Then you should be presented with a button that allows you to just accept the change.

We will definitely want to use this feature a lot.